### PR TITLE
Add Percona XtraDB Constants

### PR DESCRIPTION
**Pull request by spectro30**
Signed-off-by: Md. Alif Biswas <alif@appscode.com>

### CODE REVIEW

File: apis/kubedb/v1alpha2/constants.go [hunk 1]
```diff
@@ -214,32 +214,40 @@ const (
...
+	PerconaXtraDBMetricsExporterTLSVolumeName  = "tls-metrics-exporter-config"
+	PerconaXtraDBMetricsExporterConfigPath     = "/etc/mysql/config/exporter"
 
 	// =========================== MariaDB Constants ============================
 	MariaDBMaxClusterNameLength         = 32
 	MariaDBStandaloneReplicas           = 1
 	MariaDBDefaultClusterSize           = 3
 	MariaDBDataMountPath                = "/var/lib/mysql"
-	MariaDBDataLostFoundPath            = MariaDBDataMountPath + "lost+found"
+	MariaDBDataLostFoundPath            = MariaDBDataMountPath + "/lost+found"
```
Comment (tamalsaha) [inline apis/kubedb/v1alpha2/constants.go:250]: Use `filepath.Json(MariaDBDataMountPath, "lost+found")`

File: apis/kubedb/v1alpha2/perconaxtradb_types.go [hunk 2]
```diff
@@ -126,13 +126,22 @@ type PerconaXtraDBSpec struct {
...
+// +kubebuilder:validation:Enum=server;client;exporter
 type PerconaXtraDBCertificateAlias string
 
 const (
-	PerconaXtraDBServerCert          PerconaXtraDBCertificateAlias = "server"
-	PerconaXtraDBClientCert          PerconaXtraDBCertificateAlias = "client"
-	PerconaXtraDBMetricsExporterCert PerconaXtraDBCertificateAlias = "metrics-exporter"
+	PerconaXtraDBServerCert   PerconaXtraDBCertificateAlias = "server"
+	PerconaXtraDBClientCert   PerconaXtraDBCertificateAlias = "client"
+	PerconaXtraDBExporterCert PerconaXtraDBCertificateAlias = "exporter"
```
Comment (tamalsaha) [inline apis/kubedb/v1alpha2/perconaxtradb_types.go:135]: Why are we changing metrics-exporter -> exporter?

File: apis/kubedb/v1alpha2/perconaxtradb_types.go [hunk 3]
```diff
@@ -126,13 +126,22 @@ type PerconaXtraDBSpec struct {
...
+)
+
+// +kubebuilder:validation:Enum=ca.crt;tls.crt;tls.key
+type PerconaXtraDBCertFileNameAlias string
+
+const (
+	PerconaXtraDBCACertFileName  PerconaXtraDBCertFileNameAlias = "ca.crt"
+	PerconaXtraDBTLSCertFileName PerconaXtraDBCertFileNameAlias = core.TLSCertKey
+	PerconaXtraDBTLSKeyFileName  PerconaXtraDBCertFileNameAlias = core.TLSPrivateKeyKey
```
Comment (tamalsaha) [inline apis/kubedb/v1alpha2/perconaxtradb_types.go:144]: Do we use this in an api type? If not, we don't need the enum. core.TLSCertKey and core.TLSPrivateKeyKey are already well-known constants.

File: apis/kubedb/v1alpha2/perconaxtradb_types.go (hunk 2 above)
Comment (spectro30) [reply apis/kubedb/v1alpha2/perconaxtradb_types.go:135]: Made it same as the certificate mount path `/etc/mysql/certs/exporter`

File: apis/kubedb/v1alpha2/perconaxtradb_types.go (hunk 3 above)
Comment (spectro30) [reply apis/kubedb/v1alpha2/perconaxtradb_types.go:144]: We don't use this on API type. Removed enum.

File: apis/kubedb/v1alpha2/constants.go (hunk 1 above)
Comment (spectro30) [reply apis/kubedb/v1alpha2/constants.go:250]: It's giving this error:
```
Const initializer 'filepath.Join()' is not a constant
```
reason: https://stackoverflow.com/questions/46352334/declare-constants-in-golang-using-os-getenv-results-in-const-initializer-in-os

File: apis/kubedb/v1alpha2/perconaxtradb_helpers.go [hunk 4]
```diff
@@ -307,3 +308,11 @@ func (p *PerconaXtraDB) ReplicasAreReady(lister appslister.StatefulSetLister) (b
 	expectedItems := 1
 	return checkReplicas(lister.StatefulSets(p.Namespace), labels.SelectorFromSet(p.OffshootLabels()), expectedItems)
 }
+
+func (p *PerconaXtraDB) CertMountPath(alias PerconaXtraDBCertificateAlias) string {
+	return path.Join(PerconaXtraDBCertMountPath, string(alias))
```
Comment (tamalsaha) [inline apis/kubedb/v1alpha2/perconaxtradb_helpers.go:313]: It should be filepath.Join()

Comment (spectro30) [reply apis/kubedb/v1alpha2/perconaxtradb_helpers.go]: updated

**Review by kamolhasan — APPROVED** [review]

**Pull request merged**